lib/securenative/event_manager.rb

```diff
@@ -1,158 +1,162 @@
 # frozen_string_literal: true
 
-require 'securenative/utils/secure_native_logger'
-require 'securenative/config/securenative_options'
-require 'securenative/http/securenative_http_client'
-require 'securenative/errors/securenative_sdk_error'
-require 'securenative/errors/securenative_http_error'
-
-module SecureNative
-  class QueueItem
-    attr_reader :url, :body, :retry_sending
-    attr_writer :url, :body, :retry_sending
-
-    def initialize(url, body, retry_sending)
-      @url = url
-      @body = body
-      @retry = retry_sending
-    end
+class QueueItem
+  attr_reader :url, :body, :retry_sending
+  attr_writer :url, :body, :retry_sending
+
+  def initialize(url, body, retry_sending)
+    @url = url
+    @body = body
+    @retry = retry_sending
   end
+end
 
-  class EventManager
-    def initialize(options = SecureNativeOptions.new, http_client = nil)
-      if options.api_key.nil?
-        raise SecureNativeSDKError, 'API key cannot be None, please get your API key from SecureNative console.'
-      end
+class EventManager
+  attr_reader :activated
 
-      @http_client = if http_client.nil?
-                       SecureNativeHttpClient.new(options)
-                     else
-                       http_client
-                     end
-
-      @queue = []
-      @semaphore = Mutex.new
-      @interval = options.interval
-      @options = options
-      @send_enabled = false
-      @attempt = 0
-      @coefficients = [1, 1, 2, 3, 5, 8, 13]
-
-      @thread = Thread.new { run }
+  def initialize(options = Options.new, http_client = nil)
+    if options.api_key.nil?
+      raise SecureNativeSDKError, 'API key cannot be None, please get your API key from SecureNative console.'
     end
 
-    def send_async(event, resource_path)
-      if @options.disable
-        SecureNativeLogger.warning('SDK is disabled. no operation will be performed')
-        return
-      end
+    @http_client = if http_client.nil?
+                     SecureNative::HttpClient.new(options)
+                   else
+                     http_client
+                   end
+
+    @queue = []
+    @semaphore = Mutex.new
+    @interval = options.interval
+    @options = options
+    @send_enabled = false
+    @attempt = 0
+    @coefficients = [1, 1, 2, 3, 5, 8, 13]
+
+    @activated = false
```

Comment:
Why do u need activated, if you have send_enabled seems that they are exatly the same